Bluepill-cli/Bluepill-cli/Simulator/BPSimulator.m

```diff
@@ -83,6 +91,34 @@ - (void)createSimulatorWithDeviceName:(NSString *)deviceName completion:(void (^
                        }];
 }
 
+- (void) cloneSimulatorWithDeviceSet:(SimDeviceSet *)deviceSet withDeviceName:(NSString *)deviceName completion:(void (^)(NSError *))completion {
+    // find simulator template and clone it
+    SimDevice *simulatorWithAppInstalled = [self findDeviceWithConfig:self.config andDeviceID:[[NSUUID alloc] initWithUUIDString:self.config.templateSimUDID]];
+    [BPUtils printInfo:DEBUGINFO withString:@"clone with simulator template: %@", self.config.templateSimUDID];
+    __weak typeof(self) __self = self;
+    [deviceSet cloneDeviceAsync:simulatorWithAppInstalled name:deviceName completionQueue:dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0) completionHandler:^(NSError *error, SimDevice *device){
+        __self.device = device;
+        if (__self.config.screenshotsDirectory) {
+            __self.screenshotService = [[SimulatorScreenshotService alloc] initWithConfiguration:__self.config forDevice:device];
```

Comment:
Aren't we removing SimulatorScreenshotService?

Reply:
That will be in Xcode10 release

Reply:
Shouldn't this `if(..screenshotsDirectory)` be moved to the block where there is no error? If the call to `cloneDeviceAsync` errors it seems we'll still try to call the screenshot service.